src/apps/lwaftr/ipv6_apps.lua

```diff
@@ -147,3 +150,49 @@ function Fragmenter:push ()
       end
    end
 end
+
+-- TODO: handle any NS retry policy code here
+function NDP_eth_next_hop:new(conf)
+   local o = setmetatable({}, {__index=NDP_eth_next_hop})
+   o.conf = conf
+   -- TODO: verify that the src and dst ipv6 addresses and src mac address
+   -- have been provided, in pton format.
+   o.ns_pkt = ndp.form_ns(o.aftr_mac_b4_side, o.aftr_ipv6_ip, o.next_hop_ipv6_addr)
```

Comment:
That's doable, with https://tools.ietf.org/html/rfc3122 , but requires a different approach.